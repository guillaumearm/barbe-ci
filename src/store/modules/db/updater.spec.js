const updater = require('./updater');

describe('store:db/updater', () => {
  const unknownAction = updater({ type: 'UNKNOWN' });
  const initialState = unknownAction(undefined);
  it('returns initial state', () => {
    expect(unknownAction(undefined)).toMatchSnapshot();
  });
  it('does not update state on unknown action', () => {
    expect(unknownAction(initialState)).toEqual(initialState);
  });
  describe('receive a [LOAD_DB] action', () => {
    const loadDb = (payload) => updater({ type: 'LOAD_DB', payload });
    it('does not update state when [LOAD_DB] action payload is empty', () => {
      expect(loadDb()(initialState)).toEqual(initialState);
      expect(loadDb({})(initialState)).toEqual(initialState);
    });
    it('loads db when receive a valid [LOAD_DB] action', () => {
      const newDb = { ...initialState, users: { user1: {} } }
      expect(loadDb({ db: newDb })(initialState)).toEqual(newDb);
    });
  });
});
