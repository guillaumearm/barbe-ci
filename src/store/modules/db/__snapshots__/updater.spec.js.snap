// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store:db/updater returns initial state 1`] = `
Object {
  "builds": Object {
    "entities": Object {},
    "queue": Array [],
  },
  "commits": Object {},
  "repositories": Object {},
  "tokens": Object {},
  "users": Object {},
}
`;
